src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/InputSkipIfLastAndSharedWithCase.java

```diff
@@ -0,0 +1,119 @@
+package com.puppycrawl.tools.checkstyle.checks.coding;
+
+
+public class InputSkipIfLastAndSharedWithCase
+{
+    void method(int i) {
+        switch (i) {
+            case 1:
+            default: // No violation with the new option is expected
+                break;
+            case 2:
+                break;
+        }
+
+        switch (i) {
+            case 1:
+            default: // violation Default should be last label in the case group.
+            case 2:
+                break;
+            case 3:
+                break;
+        }
+
+        switch (i) {
+            default: // violation Default should be last label in the case group.
+            case 1:
+                break;
+            case 2:
+                break;
+        }
+
+        switch (i) {
+            case 0: default: case 1: break; case 2: break;  // violation Default should be last label in the case group.
+        }
+
+        switch (i) {
+            default: case 1: break; case 2: break;  // violation Default should be last label in the case group.
+        }
+
+        switch (i) {
+            case 1: default: break; case 2: break;  // No violation with the new option is expected
+        }
+
+        switch (i) {
+            case 1:
+            default: // No violation with the new option is expected
+                break;
+            case 2:
+                break;
+            case 3:
+                break;
+        }
+
+        switch (i) {
+            case 1:
+                break;
+            default:  // violation Default should be last label in the case group.
+            case 2:
+                break;
+            case 3:
+                break;
+        }
+
+        switch (i) {
+            case 1:
+                break;
+            case 2:
+            default: // No violation with the new option is expected
+                break;
+            case 3:
+                break;
+        }
+
+        switch (i) {
+            case 1:
+                break;
+            default: // violation Default should be last label in the case group.
+            case 3:
+                break;
+            case 4:
+                break;
+        }
+
+        switch (i) {
+            case 1:
+                break;
+            case 2:
+                break;
+            default: // violation Default should be last label in the case group.
+            case 5:
+            case 6:
+                break;
+        }
+
+        switch (i) {
+            case 1:
+                break;
+            default: // violation Default should be last label in the case group.
```

Comment:
@romani Please confirm the violation that should be shown here if new option is enabled.
`default` is last case in group, but is the only one in the group and that group is not last.

IMO, the violation should still be `Default should be last label in the switch.`.
We should only use the new violation message if the group has a `case` with the `default`.